regression-test/suites/nereids_rules_p0/infer_predicate/pull_up_predicate_set_op.groovy

```diff
@@ -0,0 +1,215 @@
+    on t3.a=t.a and t3.b=t.b  order by 1,2;
+    """
+    qt_union_res """    
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<1 union select a,b from test_pull_up_predicate_set_op2 where a<1) t inner join test_pull_up_predicate_set_op3 t3
+    on t3.a=t.a and t3.b=t.b  order by 1,2; """
+
+    qt_intersect_one_side """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 intersect select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_intersect_one_side_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 intersect select a,b from test_pull_up_predicate_set_op2 ) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_except_first_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 except select a,b from test_pull_up_predicate_set_op2 where b>'ab') t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+    qt_except_second_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a>1 except select a,b from test_pull_up_predicate_set_op2 ) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+    qt_except_no_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 except select a,b from test_pull_up_predicate_set_op2 ) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_union_different_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<2 union all select a,b from test_pull_up_predicate_set_op2 where a<1) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_union_one_side_filter """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1  union select a,b from test_pull_up_predicate_set_op2 where a<1) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_union_with_const """
+    explain shape plan 
+    select t.a,t3.b from      (select a,b from test_pull_up_predicate_set_op1 where a<1 union all select a,b from test_pull_up_predicate_set_op2 where a<1 union all select 2,2) t inner join test_pull_up_predicate_set_op3 t3 
+    on t3.a=t.a and t3.b=t.b;
+    """
+
+    qt_union_all_const """explain shape plan
+    select t.a,t3.b from      (select 133333 as a,'aa' as b union all select 2,'dd' ) t inner join test_pull_up_predicate_set_op3 t3
```

Comment:
add the case for the pushing side is not a simple table but a view or a union all, etc?